src/pybind/mgr/dashboard/controllers/grafana.py

```diff
@@ -36,7 +36,14 @@ def validation(self, params):
         method = 'GET'
         url = str(Settings.GRAFANA_API_URL).rstrip('/') + \
             '/api/dashboards/uid/' + params
-        response = grafana.url_validation(method, url)
+        try:
+            response = grafana.url_validation(method, url)
+        except Exception as e:  # pylint: disable=broad-except
+            raise DashboardException(
```

Comment:
@votdev Thanks, this is a nice improvement for users to know the reason for failure.
@nizamial09 Can you also please confirm Is this handled in other certificate errors like rgw and iscsi 
refer https://tracker.ceph.com/issues/45937 ?

Reply:
I checked the iSCSI implementation too because we stumbled upon the same problem there, but the REST API endpoints are acting very different here compared with Grafana and the changes of this PR. The iSCSI endpoints ignore connection errors. I think this is by intention, because such gateways are marked as `down`. Throwing an exception here will break the whole workflow which will lead in an empty list, even if only one gateway of many has a connection error.

- https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/controllers/iscsi.py#L136
- https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/controllers/iscsi.py#L193